### PR TITLE
Sort SkinSection in alphabetical order

### PR DESCRIPTION
**Pull request by rednir**
Excluding `<Random Skin>` and the built-in skins, all skins are sorted alphabetically as opposed to by date added. This matches behavior from stable, and it is much easier to find skins when you have a large amount.
Also, `<Random Skin>` makes more sense to be at the very top of the list imo.

This is my first PR so I realize I may have gone about doing this in the worst way possible.

### CODE REVIEW

File: osu.Game/Overlays/Settings/Sections/SkinSection.cs [hunk 1]
```diff
@@ -112,21 +112,19 @@ private void load(OsuConfigManager config)
...
-            skinItems.Insert(firstNonDefault, random_skin_info);
-
+            skinItems = sortList(skinItems);
             skinDropdown.Items = skinItems;
         }
 
         private void itemUpdated(ValueChangedEvent<WeakReference<SkinInfo>> weakItem)
         {
             if (weakItem.NewValue.TryGetTarget(out var item))
-                Schedule(() => skinDropdown.Items = skinDropdown.Items.Where(i => !i.Equals(item)).Append(item).ToArray());
```
Comment (bdach) [inline osu.Game/Overlays/Settings/Sections/SkinSection.cs:129]: The `.Where(i => !i.Equals(item))` check has been incorrectly removed. `itemUpdated` is also fired on skin updates, not necessarily always additions.

File: osu.Game/Overlays/Settings/Sections/SkinSection.cs [hunk 2]
```diff
@@ -135,6 +133,24 @@ private void itemRemoved(ValueChangedEvent<WeakReference<SkinInfo>> weakItem)
...
+            {
+                // insert lazer built-in skins before user skins
+                if (skinsList[i].ID <= 0) {
+                    var itemToMove = skinsList[i];
+                    skinsList.RemoveAt(i);
+                    skinsList.Insert(0, itemToMove);
+                }
+            }
+            skinsList.RemoveAll(s => s.ID == SkinInfo.RANDOM_SKIN);
+            skinsList.Insert(0, random_skin_info);
```
Comment (bdach) [inline osu.Game/Overlays/Settings/Sections/SkinSection.cs:149]: Several issues with this:

* This needlessly removes three items and reinserts them back to the start again, incurring unnecessary six linear operations on the entire list, while it should only be sorting the section of the list that this cares about (so user skins). `List<T>.Sort()` has an overload that takes an index and count.
* Due to the above this also wrongly pushes the `Random Skin` option to the top, while it intentionally used to be third in the order.

**Review by bdach — CHANGES_REQUESTED** [review]
In addition to these comments, please see CI errors due to whitespace that does not adhere to our code convention.

**Comment by rednir** [comment]
Didn't use the `List<T>.Sort()` overload you mentioned because it required a parameter that kept returning a "user locale could change string comparison" error, so instead copied user skins to a new list and then sorted that. Should that be fine?

**Comment by rednir** [comment]
Seems like I still can't use git properly. Should I do something about this commit spam?

**Comment by bdach** [comment]
No need. I'm in the middle of applying changes to this branch anyways.

**Comment by bdach** [comment]
I've made a few changes to this. I think I'm personally OK with how this looks now, but would appreciate a double-check on this to be sure.

@rednir make sure to read through the changes and the commit messages to see what was changed and why.

**Review by bdach — APPROVED** [review]

**Comment by rednir** [comment]
I really appreciate the comments and your support, I'll definitely keep in mind these practices if I ever open another PR. Thanks!

**Review by smoogipoo — APPROVED** [review]

**Comment by tirAL-E** [comment]
While renaming my skin collection today, I noticed that _lowercase letters are sorted separately_ from their uppercase variants, appearing at the bottom of SkinSection.

I couldn't find this being mentioned elsewhere, and I doubt that this is intended behavior, so I'm hoping that it's fine for me to bring it up here.

https://github.com/user-attachments/assets/041c394e-a756-46ac-9fe9-db5644b9540e

The list sorting order in the video above (excluding the default options) seems to be roughly:
1. Non-Alpha characters (here: dash)
2. Uppercase letters (A-Z)
3. More non-alpha characters (here: square brackets)
4. Lowercase letters (a-z)

I did not check for other non-alpha characters (numbers, curly brackets, parantheses etc.) yet.

**Pull request merged**